packages/fluxible-router/tests/unit/lib/NavLink-test.js

```diff
@@ -25,50 +29,68 @@ var TestRouteStore = RouteStore.withStaticRoutes({
     fooAB: { path: '/foo/:a/:b', method: 'get' }
 });
 
-onClickMock = function () {
+function onClickMock() {
     testResult.onClickMockInvoked = true;
-};
+}
+
+function setup(options, done) {
+    if (options.nodeEnv) {
+        process.env.NODE_ENV = options.nodeEnv;
+    }
+    var path = fs.realpathSync(resolve.sync('../../../lib/createNavLinkComponent'));
+    delete require.cache[path];
+    path = fs.realpathSync(resolve.sync('../../../lib/NavLink'));
+    delete require.cache[path];
+
+    jsdom.env({
+        url: 'http://yahoo.com',
+        html: '<html><body></body></html>',
+        done: function (err, window) {
+            if (err) {
+                done(err);
+                return;
+            }
+            global.document = window.document;
+            global.window = window;
+            global.navigator = window.navigator;
+
+            React = require('react');
+            ReactDOM = require('react-dom');
+            ReactTestUtils = require('react-addons-test-utils');
+            var mockContext = createMockComponentContext({
+                stores: [TestRouteStore]
+            });
+            mockContext.getStore('RouteStore')._handleNavigateStart({
+                url: '/foo',
+                method: 'GET'
+            });
+            MockAppComponent = require('../../mocks/MockAppComponent');
+            NavLink = require('../../../lib/NavLink');
+            createNavLinkComponent = require('../../../lib/createNavLinkComponent');
+            testResult = {};
+            done(null, mockContext);
+        }
+    });
+}
+
+function tearDown() {
+    delete global.window;
+    delete global.document;
+    delete global.navigator;
+    process.env.NODE_ENV = ORIG_NODE_ENV;
+}
```

Comment:
refactor the logic in old `beforeEach` to be able to share between 3 test suites.